src/core/string.h

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2014-2015 Dropbox, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PYSTON_CORE_STRING_H
+#define PYSTON_CORE_STRING_H
+
+#include <stdexcept>
+
+namespace pyston {
+
+class String {
+public:
+    explicit String(const std::string& s, char* _ptr) : ptr(_ptr) {
```

Comment:
I can definitely move the copying from pyston::String to BoxedString.  I'll look at how large the changes are to removing pyston::String and replacing it with the llvm alternatives.